src/nvim/mouse.c

```diff
@@ -412,6 +412,18 @@ bool do_mouse(oparg_T *oap, int c, int dir, int count, bool fixindent)
 
   // Middle mouse button does a 'put' of the selected text
   if (which_button == MOUSE_MIDDLE) {
+    // handled outside any specific mode
```

Comment:
why? (the comment should give a hint about why, also)

Reply:
I understand that it's supposed to close the tab, despite the current mode.
AFAIK there is no specific behavior regarding middle mouse clicking on a tab in a specific mode.
Please, correct me if I'm wrong :)

Reply:
`<MiddleMouse>` can have mode-specific mappings like any other keycode. So I don't see why it ignore modes. OTOH, this might be moot, see https://github.com/neovim/neovim/pull/27522#discussion_r1493824043

Reply:
Oh, i see.
Then it would be better to enclose the "click on tab with middle mouse to close" scenario, so it would be still be able to work on mode-specific mappings.
Great catch, thank you.